framework/src/meshgenerators/MergedMeshGenerator.C

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MergedMeshGenerator.h"
+
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+
+#include "libmesh/replicated_mesh.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", MergedMeshGenerator);
+
+template <>
+InputParameters
+validParams<MergedMeshGenerator>()
+{
+  InputParameters params = validParams<MeshGenerator>();
+  params.addClassDescription("Merge multiple meshes into a single unconnected mesh.");
+  params.addRequiredParam<std::vector<MeshGeneratorName>>("inputs", "The input MeshGenerators.");
+  return params;
+}
+
+MergedMeshGenerator::MergedMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters), _input_names(getParam<std::vector<MeshGeneratorName>>("inputs"))
+{
+  // Grab the input mesh references as pointers
+  for (auto & input_name : _input_names)
+    _meshes.push_back(&getMeshByName(input_name));
+}
+
+std::unique_ptr<MeshBase>
+MergedMeshGenerator::generate()
+{
+  // Have MOOSE construct the correct libMesh::Mesh object using Mesh block and CLI parameters.
+  auto base_mesh = _mesh->buildMeshBaseObject();
+  auto mesh = dynamic_pointer_cast<UnstructuredMesh>(base_mesh);
```

Comment:
👍

Reply:
No, unfortunately 👎 . `copy_nodes_and_elements` does not work on an empty fresh mesh (`_n_parts` is zero, which leads to a division (modulo) by zero). I have not found a way around that. I'm merging everything into teh first input mesh now.